spark3/src/main/java/org/apache/iceberg/spark/source/SparkWriteBuilder.java

```diff
@@ -113,9 +113,18 @@ public BatchWrite buildForBatch() {
     Broadcast<FileIO> io = lazySparkContext().broadcast(SparkUtil.serializableFileIO(table));
     Broadcast<EncryptionManager> encryptionManager = lazySparkContext().broadcast(table.encryption());
 
-    return new SparkBatchWrite(
```

Comment:
Shall we consider some sort of `SparkWriteContext` like we have `TableScanContext`? Not related to this PR directly.

Reply:
Our writes accept a lot of common params.

Reply:
I think that's a pretty reasonable step. That would make the constructors a lot easier to maintain.

Reply:
Let me see whether we can use inner classes instead. If yes, then we don't need the write context.